src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -432,14 +432,77 @@ p2ptest_chunk(xclDeviceHandle handle, char *boptr, uint64_t dev_addr, uint64_t s
   return true;
 }
 
+static bool
```

Comment:
Add comments explaining why we need this new function for no DMA, elaborate on:

Since no DMA platforms don't have a DMA engine, we copy p2p buffer to host only buffer and run the test through m2m